### ISSUE TITLE
Adding function to translate drop-down menus

### ISSUE THREAD

**Issue by Dalekamistoso** [joncampbell123/dosbox-x#2174]
Hi :)

It would be a very good idea adding a way to translate the top bar without having the modify the code and recompile the whole thing.

In the same way as using "messages.txt" for dos translation

I would suggest a similar option that reads a text file as "menus.txt" or something similar

Thanks again for your time :)

**Comment by Wengier** [comment]
Yes, I agree this is something nice to have. Thanks for the suggestion!

**Comment by Wengier** [comment]
@Dalekamistoso I have already added the ability to translate drop-down menus. All menu texts will still be written to the standard messages.txt file, instead of a separate file. If you use Windows, the updated Windows binary package is available from:

* [dosbox-x-update.zip](http://cn-dos.fam.cx/dosware/dosbox-x-update.zip)

Just write the message file again and do the translation:-)

**Comment by Dalekamistoso** [comment]
Thank you very much :)

Can you update the link to updated build? The link you include is offline.

Also, I have a question.

I would like to contribute to the project uploading the translated files as soon as I check that are properly functional.

should I add those files in a separate repository by my own or ask for a pull request on web interface or simply send it to you?

Thanks again in advance :)

> @Dalekamistoso I have already added the ability to translate drop-down menus. All menu texts will still be written to the standard messages.txt file, instead of a separate file. If you use Windows, the updated Windows binary package is available from:
> 
> * [dosbox-x-update.zip](http://cn-dos.fam.cx/dosware/dosbox-x-update.zip)
> 
> Just write the message file again and do the translation:-)

**Comment by rderooy** [comment]
You can use the 'fork' button in the top right, to create a personal fork of the project. You can then modify that fork and once your ready you can submit your changes back.

But we don't have a location in the tree right now for translations, so I would propose we create directories as such:
```
/contrib/translations/de
/contrib/translations/es
/contrib/translations/fr
/contrib/translations/nl
```
etc...

And then inside the language directory create files like:

```
fr_BE.lng
fr_CH.lng
fr_FR.lng
```
Similar to how DOSBox staging does it.

**Comment by Dalekamistoso** [comment]
Ok, I will create those as soon I have a working build in which I can translate full menus to start the work :)

**Comment by Wengier** [comment]
@Dalekamistoso The download link should be here:

* [dosbox-x-update.zip](http://cndos.fam.cx/dosware/dosbox-x-update.zip)

Please try it again. Thanks!

**Comment by Dalekamistoso** [comment]
Great, it now works !

I will start testing the new features tomorrow!

**Comment by Dalekamistoso** [comment]
Hi!

I have completed the translation in a functional way but I found some issues in the new function:

- Two fields translated on messages.txt does not translate in dropdown menus (see attached screenshots):

![bug-001](https://user-images.githubusercontent.com/67012746/104218305-d7c95900-543c-11eb-9629-c6ce647d52e4.png)
![bug-002](https://user-images.githubusercontent.com/67012746/104218308-d861ef80-543c-11eb-8cb9-1569bd95ecc6.png)

- Translation texts are not shown in the same codepage as the rest of the messages.txt (look below)

![bug-003](https://user-images.githubusercontent.com/67012746/104218311-d8fa8600-543c-11eb-8d69-22ed321da5d5.png)

I think there is a mismatch betwen character encoding of the dropdown menus and the rest of the program at least on this SDL1 32 bits version)

At this moment I completed the translation avoiding the special characters needed (á,ó,é....) is not very nice but funcional.

I also found another bug, when I change from save slots page (cliking in any of those options inside orange rectangle, the texts return to internal english names:

![bug-004](https://user-images.githubusercontent.com/67012746/104220886-6ee3e000-5440-11eb-92bc-d5ddf094429b.png)


Thanks in advance for your time and effort :)

**Comment by Wengier** [comment]
@Dalekamistoso Thanks for reporting such issues! These items are a bit tricky to be handled, but I have already fixed them in the following Windows build, with the sole exception of special characters, which I am not quite able to reproduce (the display of special characters may depend on the code pages of both the system and the language file itself):

* [dosbox-x-update.zip](http://cndos.fam.cx/dosware/dosbox-x-update.zip)

Please check it out!

**Comment by Dalekamistoso** [comment]
@Wengier almost perfect! , 

Only one little issue left, in order to have all text chains translated. 

The text chains marked in this screenshot ignores the translation now : 

![bug-005](https://user-images.githubusercontent.com/67012746/104245741-1a9f2700-5465-11eb-8d8d-53c53da6b451.png)


Those chains are contained in the following labels:

:MENU:slot0 to :MENU:slot9

If you solve this last issue, the translation capability will be complete except the special characters, but I will do tests in the future to see which codepage uses that part and how to solve it (I think that in the future we will have to make it into a separate file as the most viable solution).

Thank you again for your infinite patience 👍

**Comment by Wengier** [comment]
@Dalekamistoso Have you tried to regenerate the language file? In the updated version there is a :EMPTY_SLOT label instead of :MENU:slot0 to :MENU:slot9. The format was slightly changed.

**Comment by Dalekamistoso** [comment]
Ooops! I missed that last change, my fault.

I already have added the new label to my translation, deleted the old ones and it works now without issue.

One last question:

Can I submit my translation file now as pull request?

Other than that you can close the this issue 🥇 or if you prefer I can close tomorrow.
